Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -1930,17 +1937,15 @@ def build_search_human_readable(args, parsed_search_results):
 
 def update_headers_from_field_names(search_result, chosen_fields):
     headers = []
-    result_keys = set()
-    for search_result_keys in search_result:
-        result_keys.update(search_result_keys.keys())
+    search_result_keys = set().union(*(d.keys() for d in search_result))  # type: Set
```

Comment:
Why was this change needed?

Reply:
Sometimes the first entry doesn't contains all the headers, make sure the get all the headers in the result.